mailbag/controller.py

```diff
@@ -35,7 +35,21 @@ def generate_mailbag(self):
         # mail_account.account_data()
         #for d in derivatives:
         #    d.do_task_per_account()
-
+        os.mkdir(os.path.join(self.args.directory, self.args.mailbag_name))
+        files = os.path.join(self.args.directory, self.args.mailbag_name, "mailbag.csv")
+        header = ['Mailbag-Message-ID', 'Message_ID', 'Email_Folder', 'Date', 'From', 'To', 'Cc', 'Bcc', 'Subject',
+                  'Content_Type', 'Error']
+        with open(files, 'w', encoding='UTF8', newline='') as f:
```

Comment:
There has to be an `if not args.dry_run` or similar before the csv is written. If it is a dry run, we probably should still build the list for each message to ensure it works. `mailbag_message_id` also has to still be created if its a dry run, as the derivatives still have to run. We should add that as a field to the model in `models.py` and add it like `message.Mailbag_Message_ID= mailbag_message_id`

Reply:
would probably be idea to write a method of building the list that's currently on line 50. Then you can call the same code either in the `with open()` loop or if its a dry run